server/src/test/java/io/crate/metadata/doc/DocTableInfoTest.java

```diff
@@ -580,4 +586,51 @@ public void test_rename_column_fixes_inner_types_of_all_its_parents() throws Exc
             .hasType(ooType);
         assertThat(table.getReference(o)).isReference().hasName("o").hasType(oType);
     }
+
+
+    /**
+     * Tests a regression introduced by https://github.com/crate/crate/commit/111ffe166e523a4a5cd3278975772ce365112b64
+     * where the number of shards of a partitioned table was not preserved when writing the table info to metadata.
+     */
+    @UseRandomizedSchema(random = false)
+    @Test
+    public void test_write_to_preserves_number_of_shards_of_partitions() throws Exception {
+        var partitionIndexName = new PartitionName(new RelationName("doc", "tbl"), singletonList("1")).asIndexName();
+        SQLExecutor e = SQLExecutor.of(clusterService)
+            .addPartitionedTable(
+                """
+                create table tbl (
+                    id int,
+                    p int
+                ) clustered into 2 shards partitioned by (p) with (number_of_replicas=0)
+                """,
+                partitionIndexName
+            );
+
+        ClusterState state = clusterService.state();
+        Metadata metadata = state.metadata();
+
+        // Change the number of shards of the partition table/template aka. ALTER TABLE tbl SET (number_of_shards=3)
+        var oldTemplate = metadata.templates().get(PartitionName.templateName("doc", "tbl"));
+        var newTemplate = new IndexTemplateMetadata.Builder(oldTemplate);
+        newTemplate.settings(Settings.builder().put(oldTemplate.settings()).put("index.number_of_shards", 3));
+
+        Metadata.Builder builder = new Metadata.Builder(metadata);
+        builder.put(newTemplate);
+        var newMetadata = builder.build();
+
+        DocTableInfoFactory docTableInfoFactory = new DocTableInfoFactory(e.nodeCtx);
+        DocTableInfo tbl = docTableInfoFactory.create(RelationName.fromIndexName("tbl"), newMetadata);
+
+        var newBuilder = new Metadata.Builder(metadata);
+        tbl.writeTo(imd -> MapperTestUtils.newMapperService(
```

Comment:
Had t ocreate a new mapper service, since the `writeTo()` signature is different in 5.7